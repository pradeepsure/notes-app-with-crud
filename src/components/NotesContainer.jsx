import React, { Fragment } from 'react';
import GridView from './GridView';
import ListView from './ListView';

const NotesContainer = (props) => ({
    render() {
        const {myNotes,handleDeleteNote} = this.props;

        return (
            <Fragment>
                {(myNotes.length > 0) ?
                    ((this.props.viewType === 'gridView') ?
                        (<GridView myNotes={myNotes} handleDeleteNote={handleDeleteNote} />) : (<ListView myNotes={myNotes} handleDeleteNote={handleDeleteNote} />))
                    : (<p align='center'><h2>Your Notes Empty!!</h2></p>)}
                
            </Fragment>
        );
    }
});

export default NotesContainer;