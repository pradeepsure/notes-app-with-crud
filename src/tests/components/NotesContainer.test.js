import NotesContainer from '../../components/NotesContainer';
import React from 'react';
import { shallow, mount } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));


let wrapper, notes, removeNote;
beforeEach(() => {    
    notes= [{
        "noteTitle": "Freecharger",
        "noteDescription": "Discount mela",
        "id": 1
      }]
      removeNote = jest.fn();
    wrapper = shallow(
        <NotesContainer notes={notes} removeNote={removeNote} />
    );
});


test('Render NotesContainer component correctly', () => {        
    expect(wrapper).toMatchSnapshot();
});