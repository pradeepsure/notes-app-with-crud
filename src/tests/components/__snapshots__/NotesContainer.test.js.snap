// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NotesContainer component correctly 1`] = `
<WithStyles(Grid)
  container={true}
  spacing={8}
>
  <WithStyles(Grid)
    item={true}
    key="1"
    sm={4}
    xl={3}
    xs={12}
  >
    <WithStyles(Note)
      note={
        Object {
          "id": 1,
          "noteDescription": "Discount mela",
          "noteTitle": "Freecharger",
        }
      }
    />
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
